python/ucxx/__init__.py

```diff
@@ -6,6 +6,17 @@
 import logging
 import os
 
+# If libucx was installed as a wheel, we must request it to load the library symbols.
+# Otherwise, we assume that the library was installed in a system path that ld can find.
+try:
+    import libucx
+except ModuleNotFoundError:
+    pass
+else:
+    libucx.load_library()
+    del libucx
```

Comment:
This is a bit unexpected to my eyes. Why won't the libraries be found when loading the shared libraries with wheels?

Reply:
Yes, this is something I'm introducing in the new wheels. The idea is that wheels don't install to a standard location, so the loader doesn't know where those libraries are. The only way you could make them be found is by doing something like setting the RPATH. RPATH modification is fragile, though, especially across package boundaries. It relies on specific file layouts etc that we cannot guarantee. The approach here basically makes it so that each Python package is responsible for loading the libraries that it bundles (basically via dlopen) so that they're visible to future libraries. That way, we don't need to deal with cross-package RPATH hacking; instead, each package is responsible for loading its own libraries and consumers just call `load_library` like this.

Reply:
I pointed @pentschev to some resources offline, but tl;dr yes this is a potential problem, but this is about as good of a solution as we're going to get if we go the dynamic linking route. The only way to completely avoid these issues is with static linking, which we're trying to move away from in RAPIDS in general due to issues like binary size considerations (and also that isn't a viable solution for UCX anyway due to its extensive usage of dlopen and the way the library itself is written requiring a dynamic cudart).